### PR TITLE
[api] add getStorageAt method on web3

### PR DESCRIPTION
**Pull request by Liuhaai**

### CODE REVIEW

File: api/web3server.go [hunk 1]
```diff
@@ -725,6 +726,34 @@ func (svr *Web3Server) getTransactionByBlockNumberAndIndex(in interface{}) (inte
...
+	pos, err := hex.DecodeString(removeHexPrefix(storagePos))
+	if err != nil {
+		return nil, err
+	}
+	val, err := svr.coreService.ReadContractStorage(context.Background(), contractAddr, pos)
+	if err != nil {
+		return nil, err
+	}
+	return "0x" + hex.EncodeToString(val), nil
+}
```
Comment (dustinxie) [inline api/web3server.go:755]: can you add a test?
1. deployContractV2()
2. read this contract pos = 0x0, it should return 32-byte all 0

**Review by dustinxie — APPROVED** [review]

File: api/web3server.go [hunk 2]
```diff
@@ -725,6 +726,34 @@ func (svr *Web3Server) getTransactionByBlockNumberAndIndex(in interface{}) (inte
...
+	}
+	storagePos, err := getStringFromArray(in, 1)
+	if err != nil {
+		return nil, err
+	}
+	ioAddr, err := ethAddrToIoAddr(ethAddr)
+	if err != nil {
+		return nil, err
+	}
+	contractAddr, err := address.FromString(ioAddr)
```
Comment (CoderZhi) [inline api/web3server.go:742]: address.FromHex

File: api/web3server_utils.go [hunk 3]
```diff
@@ -410,6 +410,14 @@ func byteToHex(b []byte) string {
 	return "0x" + hex.EncodeToString(b)
 }
 
+func hexToByte(str string) ([]byte, error) {
+	str = removeHexPrefix(str)
+	if len(str)%2 == 1 {
```
Comment (CoderZhi) [inline api/web3server_utils.go:415]: could this happen?

File: api/web3server_utils.go [hunk 4]
```diff
@@ -410,6 +410,14 @@ func byteToHex(b []byte) string {
 	return "0x" + hex.EncodeToString(b)
 }
 
+func hexToByte(str string) ([]byte, error) {
```
Comment (CoderZhi) [inline api/web3server_utils.go:413]: hexToBytes

**Review by CoderZhi — APPROVED** [review]

File: api/web3server_utils.go (hunk 3 above)
Comment (Liuhaai) [reply api/web3server_utils.go:415]: yes

**Pull request merged**